File: roadmap-exercises/two-sum.go
func twoSum(nums []int, target int) []int {
	for index, current := range nums {
        difference := target - current
        if indexDifference := indexOf(nums, difference); index != indexDifference && indexDifference >= 0 {
			return []int{index, indexDifference}
		}
	}
	return []int{0, 0}
}

func indexOf(arr []int, value int) int {
	for index, current := range arr {
		if current == value {
			return index
		}
	}
	return -1
}

File: roadmap-exercises/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        prevMap = {}

        for i, n in enumerate(nums):
            difference = target - n
            if difference in prevMap:
                return [prevMap[difference], i]
            prevMap[n] = i

        return []